apps/question-bank/src/main/java/gov/cdc/nbs/questionbank/addtab/repository/WaUiMetaDataRepository.java

```diff
@@ -10,4 +10,7 @@ public interface WaUiMetaDataRepository extends JpaRepository<WaUiMetadata, Long
 
     @Query(value = "SELECT MAX(w.order_nbr) FROM WA_Ui_metadata w WHERE w.wa_template_uid = ?1 ", nativeQuery = true)
     Long findMaxOrderNumberByTemplateUid(Long waTemplateId);
+    @Query(value = "SELECT w.order_nbr FROM WA_Ui_metadata w WHERE w.wa_ui_metadata_uid = ?1 ", nativeQuery = true)
+    Long findOrderNbrForSection(Long waUiMetadataUid);
+
```

Comment:
Its better to avoid nativeQuery's wherever possible. Hibernate can do some optimizations on JPQL queries that it cant on native queries. -- That being said, this query is so simple it won't make much difference but its still nice to try to avoid nativeQueries when you can.

Reply:
Oh, and this method name could be a little more generic as well. Since it doesn't use a `nbs_ui_component_uid` in the query, it could just be `findOrderNbr`. This would make it look cleaner when re-using this to find the order nbr for other wa_ui_metatadata objects.